common/p_map.cpp

```diff
@@ -697,12 +697,31 @@ static BOOL PIT_CheckThing (AActor *thing)
 		// [SL] Work-around the additional height added to players
 		// in P_CheckPosition. Don't let players grab items above
 		// their real height!
-		if (!P_AllowPassover() || !tmthing->player ||
-			thing->z < tmthing->z + tmthing->height - 24*FRACUNIT)
+		if (!P_AllowPassover() || thing->z < tmthing->z + tmthing->height - 24*FRACUNIT)
```

Comment:
Since we're removing the check for `tmthing->player` here, I think it would be good to add a `max_z` variable as part of this check.
```
fixed_t max_z = tmthing->z + tmthing->height;
if (tmthing->player)
    max_z -= 24 * FRACUNIT;
if (!P_AllowPassover() || thing->z < max_z)
    P_TouchSpecialThing(thing, tmthing);
```

Reply:
Fixed.